core/graph/src/main/java/uk/gov/gchq/gaffer/graph/library/FileGraphLibrary.java

```diff
@@ -0,0 +1,120 @@
+        }
+        this.path = path;
+    }
+
+    @Override
+    public Pair<String, String> getIds(final String graphId) {
+        Pair<String, String> ids;
+
+        if (getGraphsPath(graphId).toFile().exists()) {
+            try {
+                List<String> lines = Files.readAllLines(getGraphsPath(graphId));
+                String[] split = lines.get(0).trim().split(",");
+                ids = new Pair<>(split[0], split[1]);
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Could not read graphs file: " + getGraphsPath(graphId), e);
+            }
+        } else {
+            return null;
+        }
+        return ids;
+    }
+
+    @Override
+    protected void _addIds(final String graphId, final Pair<String, String> schemaAndPropsIds) throws OverwritingException {
+        String schemaAndPropsIdsString = new String(schemaAndPropsIds.getFirst() + "," + schemaAndPropsIds.getSecond());
+        try {
+            FileUtils.writeStringToFile(getGraphsPath(graphId).toFile(), schemaAndPropsIdsString);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Could not write Graphs to path: " + getSchemaPath(graphId), e);
+        }
+    }
+
+    @Override
+    protected void _addSchema(final String schemaId,
+                              final byte[] schema) throws OverwritingException {
+        try {
+            FileUtils.writeByteArrayToFile(getSchemaPath(schemaId).toFile(), schema);
+        } catch (final IOException e) {
+            throw new IllegalArgumentException("Could not write schema to path: " + getSchemaPath(schemaId), e);
+        }
+    }
+
+    @Override
+    protected void _addProperties(final String propertiesId,
+                                  final StoreProperties properties) {
+        try (FileOutputStream propertiesFileOutputStream = new FileOutputStream(getPropertiesPath(propertiesId).toFile())) {
+            properties.getProperties().store(propertiesFileOutputStream, null);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Could not write properties to path: " + getSchemaPath(propertiesId), e);
+        }
+    }
+
+    @SuppressFBWarnings(value = "PZLA_PREFER_ZERO_LENGTH_ARRAYS", justification = "null represents there is no schema")
+    @Override
+    protected byte[] _getSchema(final String graphId) {
+        final Path path = getSchemaPath(graphId);
+        try {
+            return path.toFile().exists() ? Files.readAllBytes(path) : null;
+        } catch (IOException e) {
+            throw new SchemaException("Unable to read schema bytes from file: " + getSchemaPath(graphId));
+        }
+    }
+
+    @Override
+    protected StoreProperties _getProperties(final String propertiesId) {
+        return StoreProperties.loadStoreProperties(getPropertiesPath(propertiesId));
```

Comment:
You will need to put a null check in to check the file getPropertiesPath(propertiesId) actually exists and has properties in. A user could provide any propertiesId they want and it might not exist.